end-to-end-tests/tst/helpers/resources.py

```diff
@@ -0,0 +1,129 @@
+
+    def add(self, logical_resource_id=None, resource_type=None, resource_status=None, stack_resource=None):
+        if stack_resource is not None:
+            # LogicalID of ApiGateway::Deployment resource gets suffixed with SHA of the Swagger URI.
+            # This suffix changes on every run of the test. To keep tests simple, we will validate and strip
+            # out this suffix here. Tests can continue to access and verify the resource using the LogicalId
+            # specified in the template
+            logical_id = self.clean_logicalid_if_necessary(
+                logical_id_to_clean=stack_resource.get("LogicalResourceId"),
+                resource_type=stack_resource.get("ResourceType"),
+            )
+            return self.add(
+                logical_resource_id=logical_id,
+                resource_type=stack_resource.get("ResourceType"),
+                resource_status=stack_resource.get("ResourceStatus"),
+            )
+        else:
+            new_resource = Resource(logical_resource_id, resource_type, resource_status)
+            if logical_resource_id in self.resources.keys():
+                raise ValueError("Resource already exists. LogicalID = {}".format(logical_resource_id))
+            self.resources[logical_resource_id] = new_resource
+            return self
+
+    def set_status(self, logical_resource_id, new_status):
+        resource = self.resources.get(logical_resource_id)
+
+        if resource is None:
+            raise ValueError("Resource with logical ID not found - {}".format(logical_resource_id))
+        resource.status = new_status
+        return self
+
+    def remove(self, logical_resource_id):
+        self.resources.remove(logical_resource_id)
+        return self
+
+    def from_list(self, resources_list):
+        resources_from_list = Resources()
+        for stack_resource in resources_list:
+            resources_from_list.add(stack_resource=stack_resource)
+        return resources_from_list
+
+    # helper method for adding tests, call when needed
+    def get_changed_resources(self, expected_resources):
+        diff_resources = dict()
+        for key in self.resources.keys():
+            if key not in expected_resources.resources or not expected_resources.resources.get(
+                key
+            ) == self.resources.get(key):
+                diff_resources[key] = self.resources.get(key)
+            # remove the resource if its same in generated resources and expected resources
+            expected_resources.resources.pop(key, None)
+
+        if expected_resources.resources is not None:
+            diff_resources.update(expected_resources.resources)
+
+        return diff_resources
+
+    def __eq__(self, other):
+        if not isinstance(other, Resources):
+            return False
+        # checking if the keys are same
+        if self.resources.keys() != other.resources.keys():
```

Comment:
Can't we just compare the dictionaries? ` self.resources == other.resources`?

Reply:
Thank you, updated the code to just compare the dictionaries directly and it works as expected